code/game/objects/structures/table_frames.dm

```diff
@@ -20,47 +20,81 @@
 	max_integrity = 100
 	var/framestack = /obj/item/stack/rods
 	var/framestackamount = 2
+	var/list/allowed_materials
+
+	var/list/material_to_table_type = list(
+		/obj/item/stack/tile/carpet = /obj/structure/table/wood/fancy,
+		/obj/item/stack/tile/carpet/twenty = /obj/structure/table/wood/fancy,
```

Comment:
With the inclusion of `/twenty` and such, I'm not 100% sure this is the way... I want others to weigh in on this to hear what they think.

If it weren't for the base types being actual non-abstract objects such as `/obj/item/stack/tile/carpet`. Then you could perhaps use a modified version of `typecacheof` which would make a typecache per stack type for a table.

Or if the `/number` ones were spawners instead of subtypes

Reply:
I'm not gonna lie, you asked me to do it this way in your last review. I've already basically rewritten this PR once, please don't suggest I change something to `x` and then turn around to say you don't want `x`

Reply:
You can have stacks control it or you can have a bigass map like this. Changing all carpet subtypes to spawners is well out of scope and I am not doing that in this PR @farie82

Reply:
I'm a bit out of the loop on what was requested, but this does seem odd to have to include subtypes in this way. But due to this being a typecache expecting a strict type, I'm not sure if there's a way around that.

Reply:
So yeah, a further discussion was held on discord. There is either: a) have stacks control what table they make with a variable, or b) this big list that controls things

Reply:
Gotta say, I probably prefer the variable on the carpet. That way, we don't have to write these massive lists and include every subtype under the sun. But, I realize that's what you just refactored it away from... 😔

Reply:
Alright @SteelSlayer @farie82 I've reverted to the pre-decoupled method.